api/middlewares/serve-index.js

```diff
@@ -0,0 +1,17 @@
+const { join } = require('path');
+
+const ASSETS_PATH = join(__dirname, '../../assets');
+
+function serveIndex(req, res, next) {
+  try {
+    const isFile = req.url.startsWith('/assets') || req.url.includes('.');
+    if (isFile) {
+      return next();
+    }
```

Comment:
@itsalaidbacklife this is not the right way to do this, but I was proofing it out. As far as I can tell we _need_ to use custom middleware, or move the index.html file to a sails.js view.

I don't think that will work either though, because vite requires the index.html in the root so we'd need some sort of post build action to move the file which feels way weirder.

I'd like to talk to the sails maintainers and see if there's a better recommendation for a "catch all" route that a SPA would typically use.

This paves the way for the web router, and also would eventually allow us to move to vite as a middleware to consolidate the need to run two apps so that we would only need to run one (which would be way nicer).